srv/wordpress/wp-content/plugins/acore-wp-plugin/src/Hooks/WooCommerce/FieldElements.php

```diff
@@ -46,17 +48,28 @@ public static function charList($username, $deleted = false): void {
                         $bannedChars[] = $value->getName();
                         continue;
                     }
-                    echo '<option value="' . $value->getGuid() . '">' . $value->getName() . '</option>';
+                    echo '<option value="' . $value->getGuid() . '" data-charicon="' . ACORE_URL_PLG . "web/assets/race/" . $value->getRace() . ($value->getGender() == 0 ? "m" : "f") . ".webp" . '" data-classicon="' . ACORE_URL_PLG . "web/assets/class/" . $value->getClass() . ".webp" . '">' . $value->getName() . '</option>';
                 endforeach;
             } else {
                 foreach ($deletedCharacters as $key => $value):
-                    echo '<option value="' . $value->getGuid() . '">' . $value->getDeletedName() . '</option>';
+                    echo '<option value="' . $value->getGuid() . '" data-charicon="' . ACORE_URL_PLG . "web/assets/race/" . $value->getRace() . ($value->getGender() == 0 ? "m" : "f") . ".webp" . '" data-classicon="' . ACORE_URL_PLG . "web/assets/class/" . $value->getClass() . ".webp" . '">' . $value->getDeletedName() . '</option>';
                 endforeach;
             }
 
             ?>
         </select>
         <br>
+        <script>
+            function setIcon() {
+                var charicon = document.getElementById("char-icon");
+                var classicon = document.getElementById("class-icon");
+                var selected = jQuery(this).find("option:selected");
+                charicon.src = selected.data("charicon");
+                classicon.src = selected.data("classicon")
```

Comment:
minor, it's missing a semicolon
```suggestion
                classicon.src = selected.data("classicon");
```